pkg/controller/installation/products/launcher/reconciler.go

```diff
@@ -0,0 +1,298 @@
+	}
+
+	phase, err = r.reconcileComponents(ctx, serverClient)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	phase, err = r.reconcileRHSSOIntegration(ctx, serverClient)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	logrus.Infof("%s is successfully reconciled", defaultLauncherName)
+
+	return phase, err
+}
+
+func (r *Reconciler) reconcileComponents(ctx context.Context, serverClient pkgclient.Client) (v1alpha1.StatusPhase, error) {
+
+	cr := &launcherv1alpha2.Launcher{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      defaultLauncherName,
+			Namespace: r.Config.GetNamespace(),
+		},
+		Spec: launcherv1alpha2.LauncherSpec{
+			OpenShift: launcherv1alpha2.OpenShiftConfig{
+				ConsoleURL: r.installation.Spec.MasterURL,
+			},
+			OAuth: launcherv1alpha2.OAuthConfig{
+				Enabled: true,
+			},
+		},
+	}
+
+	err := serverClient.Get(ctx, pkgclient.ObjectKey{Name: cr.Name, Namespace: cr.Namespace}, cr)
+	if err != nil && !k8serr.IsNotFound(err) {
+		return v1alpha1.PhaseFailed, err
+	}
+	if err != nil {
+		if err := serverClient.Create(ctx, cr); err != nil {
+			return v1alpha1.PhaseFailed, pkgerr.Wrap(err, "failed to create launcher custom resource during reconcile")
+		}
+	}
+
+	// No status is available in the Launcher custom resource, will need to check deploymentconfigs to ensure installation is ready
+	launcherDcs, err := r.appsv1Client.DeploymentConfigs(r.Config.GetNamespace()).List(metav1.ListOptions{LabelSelector: "app=fabric8-launcher"})
+	if err != nil {
+		return v1alpha1.PhaseFailed, pkgerr.Wrap(err, "failed to list deployment configs")
+	}
+	dcStatus := olm.GetDeploymentConfigStatus(launcherDcs.Items)
+	if len(dcStatus.Starting) == 0 && len(dcStatus.Stopped) == 0 && len(dcStatus.Ready) > 0 {
+		logrus.Infof("%s application is ready", defaultLauncherName)
+		return v1alpha1.PhaseCompleted, nil
+	}
+
+	return v1alpha1.PhaseInProgress, nil
+}
+
+func (r *Reconciler) reconcileRHSSOIntegration(ctx context.Context, serverClient pkgclient.Client) (v1alpha1.StatusPhase, error) {
+	rhssoConfig, err := r.ConfigManager.ReadRHSSO()
+	if err != nil {
+		return v1alpha1.PhaseFailed, err
+	}
+	rhssoNamespace := rhssoConfig.GetNamespace()
+	rhssoRealm := rhssoConfig.GetRealm()
+	if rhssoNamespace == "" || rhssoRealm == "" {
+		logrus.Info("Cannot configure SSO integration without SSO namespace and SSO realm")
+		return v1alpha1.PhaseInProgress, nil
+	}
+
+	kcr := &aerogearv1.KeycloakRealm{}
+	err = serverClient.Get(ctx, pkgclient.ObjectKey{Name: rhssoRealm, Namespace: rhssoNamespace}, kcr)
+	if err != nil {
+		return v1alpha1.PhaseFailed, err
+	}
+
+	launcherRoute := &routev1.Route{}
+	err = serverClient.Get(ctx, pkgclient.ObjectKey{Name: launcherRouteName, Namespace: r.Config.GetNamespace()}, launcherRoute)
+	if err != nil {
+		return v1alpha1.PhaseFailed, pkgerr.Wrapf(err, "failed to git route for launcher")
+	}
+
+	launcherUrl := "http://" + launcherRoute.Spec.Host
```

Comment:
it's https now - https://github.com/fabric8-launcher/launcher-application/pull/913

Reply:
changes aren't on the latest release v0.0.4